packages/core/src/nuxt.js

```diff
@@ -1,43 +1,43 @@
-import https from 'https'
-import enableDestroy from 'server-destroy'
+
 import isPlainObject from 'lodash/isPlainObject'
 import consola from 'consola'
-import chalk from 'chalk'
-
-import ip from 'ip'
 
-import { Options, sequence } from '@nuxt/common'
+import { Options, Hookable, defineDeprecateGetter } from '@nuxt/common'
+import { Server } from '@nuxt/server'
 
 import { version } from '../package.json'
 import ModuleContainer from './module'
-import Renderer from './renderer'
 import Resolver from './resolver'
 
-export default class Nuxt {
+export default class Nuxt extends Hookable {
   constructor(options = {}) {
-    this.options = Options.from(options)
-
-    this.readyMessage = null
-    this.initialized = false
+    super()
 
-    // Hooks
-    this._hooks = {}
-    this.hook = this.hook.bind(this)
+    // Assign options and apply defaults
+    this.options = Options.from(options)
 
     // Create instance of core components
-    this.moduleContainer = new ModuleContainer(this)
-    this.renderer = new Renderer(this)
     this.resolver = new Resolver(this)
+    this.moduleContainer = new ModuleContainer(this)
+    this.server = new Server(this)
```

Comment:
Extending on my other comments, if you compare how you use @nuxt/server with how you'd use express then with express nuxt is a property of the server and not the otherway around (or two way). E.g. with this implementation of nuxt/server as a separate package my first thought would be that it would also be possible/needed to set something like `this.server = express()` to change the server type.
Or in other words, if core/nuxt and server are this tightly integrated why go through the trouble of splitting into separate packages ;)

Reply:
> Or in other words, if core/nuxt and server are this tightly integrated why go through the trouble of splitting into separate packages ;)

I have to say, I'm still finding it a bit too tightly coupled for the separation to make sense. But I believe it's a direction worth taking.